sapientml_core/preprocess/default/generator.py

```diff
@@ -324,7 +324,9 @@ def generate_code(self, dataset: Dataset, task: Task) -> Tuple[Dataset, Code]:
                 tpl, config=self.config, training=False, test=True, cols_japanese_text=cols_japanese_text
             )
 
-        cols_one_value_only = df.columns[df.nunique(dropna=False) == 1].tolist()
+        cols_one_value_only = dataset.training_dataframe.columns[
+            dataset.training_dataframe.nunique(dropna=False) == 1
+        ].tolist()
```

Comment:
I guess what you want to do is strictly checking for the training dataset.
For example, when a column of training dataset contains only a single value, the column is not useful for the training, so it should be removed.

Reply:
I changed lines along with your comment, and pushed again